go-controller/pkg/node/gateway_init.go

```diff
@@ -184,10 +184,15 @@ func getGatewayNextHops() ([]net.IP, string, error) {
 			}
 			gatewayIntf = defaultGatewayIntf
 		} else {
-			if gatewayIntf != defaultGatewayIntf || len(defaultGatewayNextHops) == 0 {
-				if config.Gateway.Mode == config.GatewayModeLocal && config.Gateway.AllowNoUplink {
+			if gatewayIntf != defaultGatewayIntf {
```

Comment:
looking at this a second time... I think this code is not 100% correct. If we find the defaultGatewayIntf does not match the configured gateway, we should not be using the defaultGatewayNextHops for that interface in OVN. So I should fix this to match the previous if statement logic of:

if gatewayIntf != defaultGatewayIntf || len(defaultGatewayNextHops) == 0 {

wdyt @tssurya ?

Reply:
so... its more like if the default gateway did match configured gateway then we won't enter this if but we should enter because what if default gateway nexthops is 0 right?
as in you converted the || into && which changed the logic from prior behaviour.
TBH: the if block nesting was/is a mess here :( if you could clear the clutter up.. :D

Reply:
It wasnt turned into && it just doesnt matter in the current code if the interfaces dont match. Let me push a new revision and I think you will see what I mean.

Reply:
ack I'll wait for the push, my point was:
before your change it was:
```
if gatewayIntf != defaultGatewayIntf || len(defaultGatewayNextHops) == 0 {
    if config.Gateway.Mode == config.GatewayModeLocal {
```
and after your change it became:
```
if gatewayIntf != defaultGatewayIntf {
}
if len(defaultGatewayNextHops) == 0 {
    if config.Gateway.Mode == config.GatewayModeLocal {
    }
}
```
which is not && sorry :) that was my confusion
and right, so the total logic needs to be either we have 0 nexthops OR we have a mismatch of interfaces versus what we have currently in your code is, if its an interface mismatch we don't do anything except logging the warning and doing the default gateway plumbing anyways which is wrong as we should be plumbing in the masquerade bits.
ok we are on same page.